src/Store.h

```diff
@@ -188,9 +188,11 @@ class StoreEntry : public hash_link, public Packable
     /// whether there is a corresponding locked shared memory table entry
     bool hasMemStore() const { return mem_obj && mem_obj->memCache.index >= 0; }
 
-    /// whether this is a collapsed forwarding-created public entry that still
-    /// has not received its response headers; new requests may collapse on it
-    bool collapsingInitiator() const;
+    /// whether this entry can feed collapsed requests and only them
+    bool hittingRequiresCollapsing() const;
+
+    /// allow or forbid collapsed requests feeding
+    void setCollapsingRequirement(const bool);
```

Comment:
I would name this parameter to self-document what a `true` value means. The `required` name you used in the method implementation is fine with me.

Reply:
Done.